package/MDAnalysis/core/universe.py

```diff
@@ -363,15 +371,28 @@ def __init__(self, topology=None, *coordinates, all_coordinates=False,
 
         if coordinates:
             self.load_new(coordinates, format=format, in_memory=in_memory,
-                        in_memory_step=in_memory_step, **kwargs)
+                          in_memory_step=in_memory_step, **kwargs)
 
         if transformations:
             if callable(transformations):
                 transformations = [transformations]
             self._trajectory.add_transformations(*transformations)
 
+        self._topology_atrrs = list(
+            att.attrname for att in self._topology.read_attributes)
+
+        to_guess = list(to_guess)
+        force_guess = list(force_guess)
+
+        if any(fmt in _PARSERS for fmt in self._parser) and 'MINIMAL' not in self._parser:
+            to_guess.extend(['types', 'masses'])
+
         if guess_bonds:
-            self.atoms.guess_bonds(vdwradii=vdwradii)
+            force_guess.extend(['bonds', 'angles', 'dihedrals'])
+
+        if to_guess or force_guess:
+            self.guess_TopologyAttributes(
+                context, to_guess, force_guess, vdwradii=vdwradii, ** kwargs)
```

Comment:
I know I commented on that a million times already but the comment keeps disappearing from the code.

This portion still needs to be refactored.